src/VisualStudio/Core/Def/Implementation/ProjectSystem/InvisibleEditor.cs

```diff
@@ -88,18 +89,18 @@ public InvisibleEditor(IServiceProvider serviceProvider, string filePath, IVsHie
             }
         }
 
-        private IVsProject TryGetProjectOfHierarchy(IVsHierarchy hierarchyOpt)
+        private IVsProject? TryGetProjectOfHierarchy(IVsHierarchy? hierarchy)
         {
             // The invisible editor manager will fail in cases where the IVsProject passed to it is not consistent with
             // the IVsProject known to IVsSolution (e.g. if the object is a wrapper like AbstractHostObject created by
             // the CPS-based project system). This method returns an IVsProject instance known to the solution, or null
             // if the project could not be determined.
```

Comment:
@davkean The AbtractHostObject stuff would all have been the fake IVsHierarchies you used to create for each target; that's all gone now right? I imagine we can delete this hack entirely.